src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -6019,6 +6019,16 @@ void MacroAssembler::lightweight_lock(Register obj, Register temp1, Register tem
   // instruction emitted as it is part of C1's null check semantics.
   z_lg(mark, Address(obj, mark_offset));
 
+  if (UseObjectMonitorTable) {
+    // Clear cache in case fast locking succeeds.
+    const Address om_cache_addr = Address(basic_lock, BasicObjectLock::lock_offset() + in_ByteSize((BasicLock::object_monitor_cache_offset_in_bytes())));
+    if (om_cache_addr.is_disp12()) {
+      z_mvghi(om_cache_addr, 0);
+    } else {
+      z_lghi(top, 0); // tmp1 is free at this point
+      z_stg(top, om_cache_addr);
+    }
```

Comment:
@RealLucy should I remove the else Part ? I tested on tier1 and it is not being executed, So maybe safe to remove ?

Reply:
I am often undecided myself. The code as it is now is correct for all displacements. If you omit the else part, you introduce a hidden dependency on the layout of BasicObjectLock and BasicLock. On the other hand, how likely is it that anybody will fundamentally change the layout and thus break the disp12 requirement? Compact or generally valid? **Your choice.**

Reply:
I guess let's keep it. I mean even if there is need to change the layout, then we have to remove `z_mvghi` and switch back to this implementation again. So maybe better we keep it here and hope for the best ?

Reply:
`BasicObjectLock` is a very small data structure. 12 bit offsets are more than enough. The actual offsets are 0 :-)

Reply:
I was only looking for extra safety. But I reverted it. Thanks for the suggestion.